cpp/src/arrow/adapters/orc/adapter.h

```diff
@@ -231,6 +231,19 @@ class ARROW_EXPORT ORCFileReader {
   Status NextStripeReader(int64_t batch_size, const std::vector<int>& include_indices,
                           std::shared_ptr<RecordBatchReader>* out);
 
+  /// \brief Get a stripe level record batch iterator with specified row count
+  ///         in each record batch. NextStripeReader serves as a fine grain
+  ///         alternative to ReadStripe which may cause OOM issue by loading
+  ///         the whole stripes into memory.
+  ///
+  /// \param[in] batch_size Get a stripe level record batch iterator with specified row
+  /// count in each record batch.
+  ///
+  /// \param[in] include_names the selected field names to read
+  /// \param[out] out the returned stripe reader
+  Status NextBatchReader(int64_t batch_size, const std::vector<std::string>& include_names,
```

Comment:
1) Please return `Result<RecordBatchReader>`
2) Why is this called `NextBatchReader`, not `NextStripeReader` as the others?

Reply:
To show the distinction, comparing the original  NextStripeReader  interface

Reply:
If the behavior is different, can you please update the docstring to correctly describe it?

Reply:
Also, I think you missed suggestion #1 above.

Reply:
Besides, if this method is different, it would be nice to have it tested as well, in `arrow/adapters/orc/adapter_test.cc`

Reply:
@pitrou  Why return Result<RecordBatchReader>. This follows other interfaces  convention, for example " Status NextStripeReader()" .

Reply:
@pitrou   Is docstring the interface comment? I have changed

Reply:
Also, just I have add UT to adapter_test.cc, and the result matched expectation.

Reply:
Returning `Status` and storing the actual result in a pointer-out (`T*`) parameter is the old convention. Any new APIs returning a result-or-error should return `Result<T>`. See the `ARROW_DEPRECATED` annotations above.

Reply:
> Why is this called `NextBatchReader`, not `NextStripeReader` as the others?

For the name, we could maybe also use `GetRecordBatchReader` (parquet has a similar method)

Reply:
Done